src/rpc/service_eth.go

```diff
@@ -261,3 +283,51 @@ var DefaultProcessTransaction = func(tx *txtypes.Transaction, ctx context.Contex
 	}
 	return submitTx, nil
 }
+
+func (s *EthService) WaitTillMined(ctx context.Context, tx *types.Transaction, delayInSeconds int) {
+	key, err := s.Cache.Key(tx)
+	if err != nil {
+		utils.Logger.Debug().Msgf("WaitTillMined | error in generating key | err: %v", err)
+		return
+	}
+	value := s.Cache.InclusionCache[key]
```

Comment:
Is this flag meant to indicate whether we are waiting for a specific transaction? If so, it doesn't need to store the whole transaction object, but could be just a boolean, and we could rename it a bit more explicitly so that it's easily understandable to WaitingForReceipt for example.